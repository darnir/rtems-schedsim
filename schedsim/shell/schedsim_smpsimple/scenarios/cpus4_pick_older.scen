#
#  Verify we select the oldest task at a priority
#
rtems_init 4
task_create smp1 3
task_create smp2 6
task_create smp3 7
task_create smp4 8
# GOAL: Tasks 0x0a010001 - 0x0a010004 as heir and executing
# GOAL:    on CPUs 3 - 0 respectively
cpus smp4 smp3 smp2 smp1

task_create smp5 5
clock_tick 1
# GOAL: Tasks 0x0a010001, 0x0a010002, 0x0a010003, and 0x0a010005 are
# GOAL:    heir and executing on CPUs 3 - 0 respectively
cpus smp5 smp3 smp2 smp1

task_create smp6 5
clock_tick 1
# GOAL: Tasks 0x0a010001, 0x0a010002, 0x0a010006, and 0x0a010005 are
# GOAL:    heir and executing on CPUs 3 - 0 respectively
cpus smp5 smp6 smp2 smp1

task_create smp7 5
clock_tick 1
# GOAL: Tasks 0x0a010001, 0x0a010007, 0x0a010006, and 0x0a010005 are
# GOAL:    heir and executing on CPUs 3 - 0 respectively
cpus smp5 smp6 smp7 smp1

# We should pick task smp5 (0x0a010005) on CPU 0
task_create smp8 4
# GOAL: Tasks 0x0a010001, 0x0a010008, 0x0a010006, and 0x0a010005 are
# GOAL:    heir and executing on CPUs 3 - 0 respectively
cpus smp5 smp6 smp8 smp1
