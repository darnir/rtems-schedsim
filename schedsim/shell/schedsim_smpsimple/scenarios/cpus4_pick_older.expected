Script File               : scenarios/cpus4_pick_older.scen
verbose                   : 1
==> 1: #
==> 2: #  Verify we select the oldest task at a priority
==> 3: #
==> 4: rtems_init 4
  CPU 0: Thread Heir 0x09010001 priority=255
  CPU 0: Thread Executing 0x09010001 priority=255
  CPU 1: Thread Heir 0x09010002 priority=255
  CPU 1: Thread Executing 0x09010002 priority=255
  CPU 2: Thread Heir 0x09010003 priority=255
  CPU 2: Thread Executing 0x09010003 priority=255
  CPU 3: Thread Heir 0x09010004 priority=255
  CPU 3: Thread Executing 0x09010004 priority=255
==> 5: task_create smp1 3
Creating task with:  Preemption: yes Timeslicing: no
Task (smp1) created: id=0x0a010001, priority=3
Task (smp1) starting: id=0x0a010001, priority=3
  CPU 3: Thread Heir 0x0a010001 priority=3
  CPU 3: Thread Executing 0x0a010001 priority=3
==> 6: task_create smp2 6
Creating task with:  Preemption: yes Timeslicing: no
Task (smp2) created: id=0x0a010002, priority=6
Task (smp2) starting: id=0x0a010002, priority=6
  CPU 2: Thread Heir 0x0a010002 priority=6
  CPU 2: Thread Executing 0x0a010002 priority=6
==> 7: task_create smp3 7
Creating task with:  Preemption: yes Timeslicing: no
Task (smp3) created: id=0x0a010003, priority=7
Task (smp3) starting: id=0x0a010003, priority=7
  CPU 1: Thread Heir 0x0a010003 priority=7
  CPU 1: Thread Executing 0x0a010003 priority=7
==> 8: task_create smp4 8
Creating task with:  Preemption: yes Timeslicing: no
Task (smp4) created: id=0x0a010004, priority=8
Task (smp4) starting: id=0x0a010004, priority=8
  CPU 0: Thread Heir 0x0a010004 priority=8
  CPU 0: Thread Executing 0x0a010004 priority=8
==> 9: # GOAL: Tasks 0x0a010001 - 0x0a010004 as heir and executing
==> 10: # GOAL:    on CPUs 3 - 0 respectively
==> 11: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010004 @  8 / 0x0a010004 @  8          false
  CPU 1: 0x0a010003 @  7 / 0x0a010003 @  7          false
  CPU 2: 0x0a010002 @  6 / 0x0a010002 @  6          false
  CPU 3: 0x0a010001 @  3 / 0x0a010001 @  3          false
=== End of Ready Set of Threads
==> 12: 
==> 13: task_create smp5 5
Creating task with:  Preemption: yes Timeslicing: no
Task (smp5) created: id=0x0a010005, priority=5
Task (smp5) starting: id=0x0a010005, priority=5
  CPU 0: Thread Heir 0x0a010005 priority=5
  CPU 0: Thread Executing 0x0a010005 priority=5
==> 14: clock_tick 1
ClockTick (1) ...
==> 15: # GOAL: Tasks 0x0a010001, 0x0a010002, 0x0a010003, and 0x0a010005 are
==> 16: # GOAL:    heir and executing on CPUs 3 - 0 respectively
==> 17: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010005 @  5 / 0x0a010005 @  5          false
  CPU 1: 0x0a010003 @  7 / 0x0a010003 @  7          false
  CPU 2: 0x0a010002 @  6 / 0x0a010002 @  6          false
  CPU 3: 0x0a010001 @  3 / 0x0a010001 @  3          false
=== End of Ready Set of Threads
==> 18: 
==> 19: task_create smp6 5
Creating task with:  Preemption: yes Timeslicing: no
Task (smp6) created: id=0x0a010006, priority=5
Task (smp6) starting: id=0x0a010006, priority=5
  CPU 1: Thread Heir 0x0a010006 priority=5
  CPU 1: Thread Executing 0x0a010006 priority=5
==> 20: clock_tick 1
ClockTick (1) ...
==> 21: # GOAL: Tasks 0x0a010001, 0x0a010002, 0x0a010006, and 0x0a010005 are
==> 22: # GOAL:    heir and executing on CPUs 3 - 0 respectively
==> 23: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010005 @  5 / 0x0a010005 @  5          false
  CPU 1: 0x0a010006 @  5 / 0x0a010006 @  5          false
  CPU 2: 0x0a010002 @  6 / 0x0a010002 @  6          false
  CPU 3: 0x0a010001 @  3 / 0x0a010001 @  3          false
=== End of Ready Set of Threads
==> 24: 
==> 25: task_create smp7 5
Creating task with:  Preemption: yes Timeslicing: no
Task (smp7) created: id=0x0a010007, priority=5
Task (smp7) starting: id=0x0a010007, priority=5
  CPU 2: Thread Heir 0x0a010007 priority=5
  CPU 2: Thread Executing 0x0a010007 priority=5
==> 26: clock_tick 1
ClockTick (1) ...
==> 27: # GOAL: Tasks 0x0a010001, 0x0a010007, 0x0a010006, and 0x0a010005 are
==> 28: # GOAL:    heir and executing on CPUs 3 - 0 respectively
==> 29: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010005 @  5 / 0x0a010005 @  5          false
  CPU 1: 0x0a010006 @  5 / 0x0a010006 @  5          false
  CPU 2: 0x0a010007 @  5 / 0x0a010007 @  5          false
  CPU 3: 0x0a010001 @  3 / 0x0a010001 @  3          false
=== End of Ready Set of Threads
==> 30: 
==> 31: # We should pick task smp5 (0x0a010005) on CPU 0
==> 32: task_create smp8 4
Creating task with:  Preemption: yes Timeslicing: no
Task (smp8) created: id=0x0a010008, priority=4
Task (smp8) starting: id=0x0a010008, priority=4
  CPU 2: Thread Heir 0x0a010008 priority=4
  CPU 2: Thread Executing 0x0a010008 priority=4
==> 33: # GOAL: Tasks 0x0a010001, 0x0a010007, 0x0a010006, and 0x0a010008 are
==> 34: # GOAL:    heir and executing on CPUs 3 - 0 respectively
==> 35: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010008 @  4 / 0x0a010008 @  4          false
  CPU 1: 0x0a010006 @  5 / 0x0a010006 @  5          false
  CPU 2: 0x0a010007 @  5 / 0x0a010007 @  5          false
  CPU 3: 0x0a010001 @  3 / 0x0a010001 @  3          false
=== End of Ready Set of Threads
